packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -121,24 +119,53 @@ exports.sourceNodes = async (
   // are "updated" so will get the now deleted reference removed.
 
   function deleteContentfulNode(node) {
-    const localizedNodes = locales
-      .map(locale => {
-        const nodeId = createNodeId(
-          normalize.makeId({
-            spaceId: space.sys.id,
-            id: node.sys.id,
-            currentLocale: locale.code,
-            defaultLocale,
+    locales.forEach(locale => {
+      const nodeId = createNodeId(
+        normalize.makeId({
+          spaceId: space.sys.id,
+          id: node.sys.id,
+          currentLocale: locale.code,
+          defaultLocale,
+        })
+      )
+      const gatsbyNode = getNode(nodeId)
+
+      if (gatsbyNode) {
+        Object.keys(gatsbyNode).forEach(field => {
+          // we only care about ___NODE as they are links to other ones
+          if (!field.includes(`___NODE`)) {
+            return
+          }
+
+          // remove foreignReferences to this node
+          ;[].concat(gatsbyNode[field]).forEach(value => {
```

Comment:
Is this a hack for `if (Array.isArray(gatsbyNode[field])) { .forEach` ? Can't we assume the contents of this field, or test and fail early instead?

Reply:
`gatsbyNode[field]` isn't always an array but could, so this makes sure it always is. Else I had to do a if else with duplicate code or made a function for it.